File: rust/src/models/matrix_dimension.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MatrixDimension {
    pub key: String,

    pub value: String,
}

impl MatrixDimension {
    pub fn new(key: String, value: String) -> Self {
        Self { key, value }
    }
}

File: typescript/src/models/matrixDimension.ts
// this file is @generated

export interface MatrixDimension {
  key: string;

  value: string;
}

export const MatrixDimensionSerializer = {
  _fromJsonObject(object: any): MatrixDimension {
    return {
      key: object["key"],
      value: object["value"],
    };
  },

  _toJsonObject(self: MatrixDimension): any {
    return {
      key: self.key,
      value: self.value,
    };
  },
};
